### PR TITLE
Increase Golang version and memory requirement for Vagrant test env

### PR DESCRIPTION
**Pull request by esnible**
Resolves https://github.com/istio/istio/issues/13916

### CODE REVIEW

**Comment by esnible** [comment]
@frankbu I assigned this to you because I shouldn't assign it to myself and because Google folks can't use Vagrant.  Looking for `/lgtm`.

File: tools/vagrant/provision-vagrant.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 set -e
-VERSION="1.9.2"
+VERSION="1.11.2"
```
Comment (ericvn) [inline tools/vagrant/provision-vagrant.sh]: Any particular reason you picked 1.11.x? With go module support, we may want a later version and I know KinD is saying 1.12.5 or later...

Comment (esnible) [reply tools/vagrant/provision-vagrant.sh]: 1.12.5 is the latest stable version.  I have tested with that and re-pushed.

**Pull request merged**